pallets/order-book/src/order_book.rs

```diff
@@ -688,8 +874,93 @@ impl<T: crate::Config + Sized> OrderBook<T> {
         volume
     }
 
-    fn cross_spread(&self) {
-        // todo (m.tagirov)
-        todo!()
+    pub fn cross_spread<'a>(
+        &self,
+        limit_order: LimitOrder<T>,
+        data: &mut impl DataLayer<T>,
+    ) -> Result<
+        MarketChange<
+            T::AccountId,
+            T::AssetId,
+            T::DEXId,
+            T::OrderId,
+            LimitOrder<T>,
+            BlockNumberFor<T>,
+        >,
+        DispatchError,
+    > {
+        let (market_amount, limit_amout) = match limit_order.side {
+            PriceVariant::Buy => Self::calculate_market_depth_to_price(
+                limit_order.side.switch(),
```

Comment:
Not an addition of this PR but name `switched()` should be more accurate

Reply:
`switch` as a verb in the imperative mood is more accurate

Reply:
however it does not change the state of `self` but rather returns a new value (that is actually switched)

Reply:
Ok, but it is not my code =)